pkg/transfer/local/pull_test.go

```diff
@@ -0,0 +1,189 @@
+		},
+		{
+			Name: "No match input linux/386 and defaultSnapshotter",
+			UnpackConfig: transfer.UnpackConfiguration{
+				Platform:    platforms.MustParse("linux/386"),
+				Snapshotter: containerd.DefaultSnapshotter,
+			},
+			SupportedPlatforms: supportedPlatforms,
+			Match:              false,
+			ExpectedPlatform:   transfer.UnpackConfiguration{},
+		},
+		{
+			Name: "Match linux/amd64 and native snapshotter",
+			UnpackConfig: transfer.UnpackConfiguration{
+				Platform:    platforms.MustParse("linux/amd64"),
+				Snapshotter: "native",
+			},
+			SupportedPlatforms: supportedPlatforms,
+			Match:              true,
+			ExpectedPlatform: transfer.UnpackConfiguration{
+				Platform:    platforms.MustParse("linux/amd64"),
+				Snapshotter: "native",
+			},
+		},
+		{
+			Name: "Match linux/arm64 and native snapshotter",
+			UnpackConfig: transfer.UnpackConfiguration{
+				Platform:    platforms.MustParse("linux/arm64"),
+				Snapshotter: "native",
+			},
+			SupportedPlatforms: supportedPlatforms,
+			Match:              true,
+			ExpectedPlatform: transfer.UnpackConfiguration{
+				Platform:    platforms.MustParse("linux/arm64"),
+				Snapshotter: "native",
+			},
+		},
+		{
+			Name: "Match linux/arm and defaultSnapshotter",
+			UnpackConfig: transfer.UnpackConfiguration{
+				Platform:    platforms.MustParse("linux/arm"),
+				Snapshotter: containerd.DefaultSnapshotter,
+			},
+			SupportedPlatforms: supportedPlatforms,
+			Match:              true,
+			ExpectedPlatform: transfer.UnpackConfiguration{
+				Platform:    platforms.MustParse("linux/arm"),
+				Snapshotter: containerd.DefaultSnapshotter,
+			},
+		},
+		{
+			Name: "Platform linux/amd64 input only match with defaultSnapshotter",
+			UnpackConfig: transfer.UnpackConfiguration{
+				Platform: platforms.MustParse("linux/amd64"),
+			},
+			SupportedPlatforms: supportedPlatforms,
+			Match:              true,
+			ExpectedPlatform: transfer.UnpackConfiguration{
+				Platform:    platforms.MustParse("linux/amd64"),
+				Snapshotter: containerd.DefaultSnapshotter,
+			},
+		},
+		{
+			Name: "Platform linux/arm64 input only match with defaultSnapshotter",
+			UnpackConfig: transfer.UnpackConfiguration{
+				Platform: platforms.MustParse("linux/arm64"),
+			},
+			SupportedPlatforms: supportedPlatforms,
+			Match:              true,
+			ExpectedPlatform: transfer.UnpackConfiguration{
+				Platform:    platforms.MustParse("linux/arm64"),
+				Snapshotter: containerd.DefaultSnapshotter,
+			},
+		},
+		{
+			Name: "Platform linux/arm input only match with defaultSnapshotter",
+			UnpackConfig: transfer.UnpackConfiguration{
+				Platform: platforms.MustParse("linux/arm"),
+			},
+			SupportedPlatforms: supportedPlatforms,
+			Match:              true,
+			ExpectedPlatform: transfer.UnpackConfiguration{
+				Platform:    platforms.MustParse("linux/arm"),
+				Snapshotter: containerd.DefaultSnapshotter,
+			},
+		},
+	} {
+
+		t.Run(testCase.Name, func(t *testing.T) {
```

Comment:
please use the following pattern.

```
testCase := testCase
t.Run()
```
Just in case that the testCase might be wrong if it runs in parallel.

Reply:
I did see this pattern in a few places. But didn't quite understand the reason.
With this, we create a local copy of the testCase, how could it prevent tests that might be wrong in parallel?

Reply:
`testCase := testCase` will shadow the original range var. If not, the concurrent goroutines will use that same value.